packages/core/src/cache/classes/Mesh.ts

```diff
@@ -0,0 +1,134 @@
+import vtkActor from '@kitware/vtk.js/Rendering/Core/Actor';
+import vtkMapper from '@kitware/vtk.js/Rendering/Core/Mapper';
+import vtkPLYReader from '@kitware/vtk.js/IO/Geometry/PLYReader';
+import vtkSTLReader from '@kitware/vtk.js/IO/Geometry/STLReader';
+import vtkXMLPolyDataReader from '@kitware/vtk.js/IO/XML/XMLPolyDataReader';
+import vtkOBJReader from '@kitware/vtk.js/IO/Misc/OBJReader';
+import vtkProperty from '@kitware/vtk.js/Rendering/Core/Property';
+import vtkPolyDataNormals from '@kitware/vtk.js/Filters/Core/PolyDataNormals';
+
+import type { MeshData, RGB } from '../../types';
+import { MeshType } from '../../enums';
+
+type MeshProps = MeshData;
+
+/**
+ * Mesh class for storing mesh data
+ * Each Mesh represents a single actor a 3D mesh.
+ */
+export class Mesh {
+  readonly id: string;
+  readonly sizeInBytes: number;
+  private _color: RGB = [255, 255, 255];
+  private _actors: vtkActor[] = [];
+  private _format: string;
+
+  /**
+   * Creates an instance of Mesh.
+   * @param {MeshProps} props - The properties to initialize the Mesh with.
+   */
+  constructor(props: MeshProps) {
```

Comment:
Disclosure: I'm no expert, but isn't a mesh just a rendered surface defined by points and polygons? I'm trying to understand the difference between this `Mesh` class and our `Surface` class. I see you simply create an actor and mapper, and no special VTK classes like `GridStuff` are used.

Reply:
I just wanted to separate between the segmentation stuff(surface/contour) and the Mesh, the Mesh class keeps tracks of the list of actors and their textures, a mesh is more than points and polygons i believe.
I didn't understood the last part about VTK classes, i am using the readers to parse the mesh buffer

Reply:
Ok sure makes sense, let me take a look again then